components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/transaction/store/TransactionStore.java

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+* WSO2 Inc. licenses this file to you under the Apache License,
+* Version 2.0 (the "License"); you may not use this file except
+* in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.wso2.micro.integrator.initializer.handler.transaction.store;
+
+import org.wso2.micro.integrator.initializer.handler.transaction.TransactionException;
+import org.wso2.micro.integrator.initializer.handler.transaction.store.connector.RDBMSConnector;
+
+import javax.crypto.Cipher;
+import javax.sql.DataSource;
+
+/**
+ * The layer which connects to the transaction data.
+ */
+public class TransactionStore {
```

Comment:
Shall we make this as an interface and have an RDBMS store , we can use file based store for stand alone deployments. 
ELse how are we going to handle one only one node is there , is db a must ?

Reply:
In the Kubernetes environment, each pod will consider itself as a standalone deployment. In that case, when we have a file-based store for standalone deployments and unless this feature is disabled, each pod/container will continue to store the count inside the pod where the data is not persistent. This can be an unnecessary overhead where the stored data has no actual use. How can we address that?

Reply:
True for Kubernetes. I was having VM standlone deployments in mind, how are we going to address it there

Reply:
VM standalone deployments won't have a DB. Shall we use H2 based carbon DB in that case? If a RDBMS is not present, if counting is enabled, it should default to carbon DB.

Reply:
Disabling the Transaction count component by default. Hence, if the user wants to enable this feature, the required configurations will give them the notion that a valid data source should be configured priorly during the configuration time. As IMO, this will remove the need for a default data source being configured.